spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/batch/BatchAutoConfigurationTests.java

```diff
@@ -374,6 +377,21 @@ void whenTheUserDefinesTheirOwnDatabaseInitializerThenTheAutoConfiguredBatchInit
 				.hasBean("customInitializer"));
 	}
 
+	@Test
+	void testExecuteLocalAndNonConfiguredJob() {
+		this.contextRunner
+		.withUserConfiguration(NamedJobConfigurationWithLocalJob.class, EmbeddedDataSourceConfiguration.class)
+		.withPropertyValues("spring.batch.job.names:discreteLocalJob,nonConfiguredJob")
+		.run((context) -> {
+			assertThat(context).hasSingleBean(JobLauncher.class);
+			assertThrows(NoSuchJobException.class,()->{
```

Comment:
The Spring Boot style rules don't allow JUnit `assertThrows()`. Please use the AssertJ `assertThatIllegalArgumentException()` or `assertThatExceptionOfType()` instead.